### PR TITLE
feat: add asyn iterable section in doc

### PR DESCRIPTION
**Pull request by abhishekraj272**
Issue: #991

Added a description and an exmaple of async iterables in the doc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,6 +180,36 @@ Basic usage example:
...
+
+const data = {
+  async *[Symbol.asyncIterator]() {
+    yield "hello";
+    yield "world";
+  },
+};
+
+(async () => {
+  await fetch("https://example.com", { body: data });
```
Comment (KhafraDev) [inline README.md:209]:
```suggestion
  await fetch("https://example.com", { body: data, method: 'POST' });
```
`method` property should be set here as well or else the request will fail.

File: README.md [hunk 2]
```diff
@@ -180,6 +180,36 @@ Basic usage example:
...
 ```
 
+
+#### `request.body`
+
+A body can be of the following types:
+
+- ArrayBuffer
+- ArrayBufferView
+- AsyncIterables
```
Comment (ronag) [inline README.md:190]: Can also be Iterable

Comment (abhishekraj272) [reply README.md:190]: fixed

**Review by ronag — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by abhishekraj272** [comment]
Anything left to merge this pr?

**Pull request merged**